### PR TITLE
Testing: Add Python 3.8/3.9 testing; Fix #4453

### LINKED ISSUE #4453: Python 3.8(/3.9) in Testing
Motivation
----------
Python 3.8 and 3.9 are the current feature releases of Python and compatibility of Rucio clients/server/daemons should be ensured.

Modification
------------
Modify the Dockerfile used for GitHub Actions or add a new one to test Rucio with the mentioned Python versions.

### PR DESCRIPTION
**Pull request by bziemons**
Testing: Add Python 3.8/3.9 testing; Fix #4453
Based on #4586 and #4564

### CODE REVIEW

File: etc/pip-requires-test [hunk 1]
```diff
@@ -6,7 +6,7 @@ pytest-xdist~=2.2.0; python_version >= '3.6'      # Used for parallel testing on
 pyflakes==2.1.1                                   # Passive checker of Python programs
 flake8==3.7.8                                     # Wrapper around PyFlakes&pep8
 pylint==1.9.4; python_version < '3.6'             # static code analysis. 1.9.5 last 2.7 compatible release
-pylint==2.6.0; python_version >= '3.6'
+pylint==2.8.2; python_version >= '3.6'
```
Comment (bari12) [inline etc/pip-requires-test:9]: Ah apologies, I remember I promised a PR for this too and forgot.. Thanks for adding this :-)

**Review by bari12 — DISMISSED** [review]
LGTM, you can probably re-base to remove the isAlive changes from the PR. I'll leave it open for a moment too for others to comment.

File: etc/pip-requires-test [hunk 2]
```diff
@@ -6,7 +6,7 @@ pytest-xdist~=2.2.0; python_version >= '3.6'      # Used for parallel testing on
 pyflakes==2.1.1                                   # Passive checker of Python programs
 flake8==3.7.8                                     # Wrapper around PyFlakes&pep8
```
Comment (mlassnig) [inline etc/pip-requires-test:7]: update to 3.9.2 ?

File: etc/pip-requires-test [hunk 3]
```diff
@@ -6,7 +6,7 @@ pytest-xdist~=2.2.0; python_version >= '3.6'      # Used for parallel testing on
 pyflakes==2.1.1                                   # Passive checker of Python programs
```
Comment (mlassnig) [inline etc/pip-requires-test:6]: update to 2.3.1?

**Review by mlassnig — APPROVED** [review]

File: etc/pip-requires-test (hunk 2 above)
Comment (bziemons) [reply etc/pip-requires-test:7]: see https://github.com/rucio/rucio/pull/4588

File: etc/pip-requires-test (hunk 3 above)
Comment (bziemons) [reply etc/pip-requires-test:6]: see https://github.com/rucio/rucio/pull/4588

**Pull request merged**